wandb/sdk/internal/job_builder.py

```diff
@@ -259,18 +241,54 @@ def _build_image_job_source(
 
         raw_image_name = image_name
         if ":" in image_name:
-            raw_image_name, tag = image_name.split(":")
-            self._aliases += [tag]
+            tag = image_name.split(":")[-1]
+
+            # if tag looks properly formatted, assume its a tag
+            # regex: alphanumeric add "_" "-" "."
+            if re.fullmatch(r"([a-zA-Z0-9_\-\.]+)", tag):
+                raw_image_name = raw_image_name.replace(f":{tag}", "")
+                self._aliases += [tag]
 
-        if self._settings.job_name:
-            name = self._settings.job_name
-        else:
-            name = make_artifact_name_safe(f"job-{raw_image_name}")
         source: ImageSourceDict = {
             "image": image_name,
         }
+        name = self._make_job_name(raw_image_name)
+
         return source, name
 
+    def _make_job_name(self, input_str: str) -> str:
+        """Use job name from settings if provided, else use programatic name."""
+        if self._settings.job_name:
+            return self._settings.job_name
+
+        return make_artifact_name_safe(f"job-{input_str}")
+
+    def _get_entrypoint_and_notebook(
```

Comment:
nit: This function doesn't update `self._is_notebook_run` does it need to return it? Is this for readability?